service/config_provider.go

```diff
@@ -60,6 +60,8 @@ type ConfigProvider interface {
 	//
 	// Should never be called concurrently with itself or Get.
 	Shutdown(ctx context.Context) error
+	// DryRunGet validates the configuration, prints out all related errors without running the collector
+	DryRunGet(ctx context.Context, factories component.Factories) (*Config, error, bool)
```

Comment:
Should "dry run" be a flag or a command? 

/cc @codeboten @jpkrohling who also reviewed https://github.com/open-telemetry/opentelemetry-collector/pull/6322

Reply:
i would expect it to be a flag